packages/cypress/src/add-commands/index.ts

```diff
@@ -0,0 +1,65 @@
+import type { ApplicationConfig } from '@commercetools-frontend/application-config';
+
+import { v4 as uuidv4 } from 'uuid';
+import { OIDC_RESPONSE_TYPES, STORAGE_KEYS } from '../constants';
+import { buildOidcScope } from '../helpers';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+declare const Cypress: any;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+declare const cy: any;
+
+Cypress.Commands.add(
+  'loginByOidc',
+  ({ entryPointUriPath }: { entryPointUriPath: string }) => {
+    Cypress.log({ name: 'loginByOidc' });
+
+    const projectKey = Cypress.env('PROJECT_KEY');
+    const sessionNonce = uuidv4();
+
+    cy.task('customApplicationConfig', {
+      entryPointUriPath,
+    }).then((appConfig: ApplicationConfig['env']) => {
```

Comment:
We're using a [task](https://docs.cypress.io/api/commands/task.html) to be able to perform nodejs operations, specifically using the filesystem.

Essentially we want to read up the runtime environment config (which is what is available at `window.app`) as the first thing, as we need to get some of the values to construct the request to fetch a token.